api/src/modules/import-data/sourcing-data/validators/sourcing-data.class.validator.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  IsEnum,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+  Min,
+  MinLength,
+  Validate,
+} from 'class-validator';
+import { LOCATION_TYPES } from 'modules/sourcing-locations/sourcing-location.entity';
+import { LocationAddressInputValidator } from 'modules/import-data/sourcing-data/validators/address-input.custom.validator';
+import { LocationLatitudeInputValidator } from 'modules/import-data/sourcing-data/validators/latitude-input.custom.validator';
+import { LocationLongitudeInputValidator } from 'modules/import-data/sourcing-data/validators/longitude-input.custom.validator';
+
```

Comment:
I believe this can be a bit dangerous as we get tied to the current year amount of the excel. I think it is OK to match the columns as we rely on these to build create entities, build realtions and a lot of stuff.

But the 'year columns' will 100% change, users will add more years, and most likely different users will have different years. I think (I might be wrong) that this approach will just match the current dataset

There is a sorting algorithm that finds columns including some year value and extracting both the year and value from there. I believe it would be a better idea to validate after the algorithm has dynamically sorted this out, and then recursively validate the values in these fields

Also, probably Science is validating that these columns hold a numeric value, but we should also guard the API against unexpected data.

PS: Don't trust people xD